druntime/src/core/demangle.d

```diff
@@ -394,9 +388,15 @@ pure @safe:
         size_t   tlen = 0;
         real     val  = void;
 
+        void errorReturn()
+        {
+            errStatus = true;
+        }
+
         if ( 'I' == front )
         {
-            mixin(matchOrF!( "INF" ));
+            if (!match("INF"))
+                return errorReturn();
```

Comment:
Name is not good: `errorReturn()` actually not returns control flow from `parseReal()` and not returns anything from itself

Reply:
What about
```d
errStatus = !match("INF");
if (errStatus) return;
```
?

Reply:
However, your approach also makes sense.

Reply:
I don't think it really matters, it's temporary. I just picked something unique so it's easy to rename, but I made it `onError` now.